pkg/controller/cluster/cluster_controller.go

```diff
@@ -348,6 +345,80 @@ func (c *clusterController) reconcileHostCluster() error {
 	return err
 }
 
+func (c *clusterController) judgeIfClusterIsReady() error {
+	clusters, err := c.clusterLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	for _, cluster := range clusters {
+		if len(cluster.Spec.Connection.KubeConfig) == 0 {
+			continue
+		}
+
+		clientConfig, err := clientcmd.NewClientConfigFromBytes(cluster.Spec.Connection.KubeConfig)
+		if err != nil {
+			klog.Error(err)
+			continue
+		}
+
+		config, err := clientConfig.ClientConfig()
+		if err != nil {
+			klog.Error(err)
+			continue
+		}
+		config.Timeout = 10 * time.Second
```

Comment:
What if there are lots of clusters, saying 50, and each cluster takes 9s to finish probing, that would be 450s, > resyncPeriod.

Reply:
10 seconds seem to be too long. But the case you are talking about is very rare. It's unlikely that every cluster connection takes 9s. In most cases, one connection takes several ms. How about change the timeout to 3s?

Reply:
If there are network issues on the node where the `ks-controller-manager` pod residing, it's possible.

Reply:
What I did before put cluster back to working queue every resyncPeriod, and check its readiness on main sync loop.

Reply:
I didn't see ` put cluster back to working queue every resyncPeriod`, but I saw `check its readiness on main sync loop.`. I think we don't need to `put cluster back to working queue every resyncPeriod` manually, the cluster informer does that automatically. The reason I check the cluster readiness separately is to check all of the cluster readiness, not only proxy connection. What you did before only checks the proxy cluster if has `agent availably` status, then updates the cluster status to ready or not. By using kubeconfig, I think it's more reliable.

- [ ]

Reply:
If put in the main sync loop, once the cluster is created/updated/deleted, the check will be performed, which may be too frequent. On the other hand, the check may be too long and will affect the sync loop. what is your suggestion?

Reply:
That's true, but we need to update `cluster.status.configz` every resyncPeriod too. So I suggest make `config.timeout` shorter, and probe in the main loop.

Reply:
Currently, we update the `cluster.status.configz` every resyncPeriod at the end main loop. The update doesn't change.

Reply:
OK, better to make `config.timeout` shorter

Reply:
`config.Timeout` has been set as 3s by default. We can merge this pr now.